features/graph/api/src/main/java/org/opennms/netmgt/graph/api/generic/AllowedValuesInPropertiesMap.java

```diff
@@ -0,0 +1,116 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2019-2019 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2019 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.graph.api.generic;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import org.opennms.netmgt.graph.api.info.NodeInfo;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+
+/** 
+ * This class holds all allowed types that can be used inside the properties of a GenericElement. They need to be immutable and persistable.
+ * See also org.opennms.netmgt.graph.persistence.converter.ConverterService
+ * TODO: Patrick: discuss with mvr if and how we want to synchronize this list with the converters in converter service. 
+ */
+public class AllowedValuesInPropertiesMap {
+
+	private final static List<AllowedValue> ALLOWED_VALUES = new ArrayList<>();
+	static {
+		ALLOWED_VALUES.add((o)->o instanceof Boolean);
+		ALLOWED_VALUES.add((o)->o instanceof Float);
+		ALLOWED_VALUES.add((o)->o instanceof Integer);
+		ALLOWED_VALUES.add((o)->o instanceof Double);
+		ALLOWED_VALUES.add((o)->o instanceof String);
+		ALLOWED_VALUES.add((o)->o instanceof Short);
+		ALLOWED_VALUES.add((o)->o instanceof Byte);
+		ALLOWED_VALUES.add((o)->o instanceof Enum);
+		ALLOWED_VALUES.add(new AllowedValueCollection());
+		// TODO: Patrick make sure NodeInfo can also be persisted, discuss with mvr the strategy
```

Comment:
@mvrueden NodeInfo have no converter so far. Serialize as JSon?

Reply:
Rather than a static initialization block, it would be a bit cleaner to do:

`private final static List<AllowedValue> ALLOWED_VALUES = Arrays.asList(...);`

Reply:
agree. Fixed.

Reply:
`NodeInfo` is a calculated/enriched value and represents a node. It should never be persisted, as the data is already in the database. If a node reference must be persisted, then use the according `NodeRef` object instead, which is a simple String -> `<foreignSource>:<foreignId>`.